src/osd/scheduler/mClockScheduler.cc

```diff
@@ -30,6 +30,73 @@ using namespace std::placeholders;
 
 namespace ceph::osd::scheduler {
 
+static void log_mclock_queue_stats(PerfCounters *logger, scheduler_id_t &id, bool is_add)
+{
+  if (!logger) {
+    return;
+  }
+
+  switch (id.class_id) {
+  case op_scheduler_class::immediate:
+    if (is_add) {
+      logger->inc(l_mclock_immediate_queue_len);
+    } else {
+      logger->dec(l_mclock_immediate_queue_len);
+    }
+    break;
+  case op_scheduler_class::client:
+    if (is_add) {
+      logger->inc(l_mclock_client_queue_len);
+    } else {
+      logger->dec(l_mclock_client_queue_len);
+    }
+    break;
+  case op_scheduler_class::background_recovery:
+    if (is_add) {
+      logger->inc(l_mclock_recovery_queue_len);
+    } else {
+      logger->dec(l_mclock_recovery_queue_len);
+    }
+    break;
+  case op_scheduler_class::background_best_effort:
+    if (is_add) {
+      logger->inc(l_mclock_best_effort_queue_len);
+    } else {
+      logger->dec(l_mclock_best_effort_queue_len);
+    }
+    break;
+   default:
+    // should assert
+    break;
+  }
+}
```

Comment:
This seems unnecessarily complicated, how about 

```
unsigned get_mclock_counter(scheduler_id_t id) { ... }


logger->inc(get_mclock_counter(id));
```

That avoid needing to pass logger or is_add (which appears to be passed statically from all callers) and makes it much more obvious what's happening.

Reply:
Idea seems reasonable, though.  @sseshasa thoughts?